21-GitTfsShelvesets/readme.md

```diff
@@ -0,0 +1,37 @@
+Git TFS Shelvesets
+==================
+
+We can work with TFS shelvesets from git.
+
+Listing existing shelvesets
+---------------------------
+
+List shelvesets, by default just yours using this command
+
+	$ git tfs shelve-list
+
+You can use:
+
+* `--full` to get more informations
+* `-u=all` to list shelvesets of all users, not just you
+
+> I have no idea what happens when you have user named `all`.
+
+Unshelve
+--------
+
+	$ git [-u=user] tfs unshelve shelveset_name branch_name
+
+Unshelves shelveset onto new branch, which is automatically created from parent commit of the shelveset.
+
+Shelve
+------
+
+	$ git tfs shelve shelveset_name
```

Comment:
I found very useful here -m 'TFS message' here - otherwise you have as comment all comments from git history which is perhaps not ideal